src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -641,6 +650,7 @@ public synchronized void initServer(int delay) throws ConfigurationException
         logger.info("Thrift API version: {}", cassandraConstants.VERSION);
         logger.info("CQL supported versions: {} (default: {})",
                     StringUtils.join(ClientState.getCQLSupportedVersion(), ","), ClientState.DEFAULT_CQL_VERSION);
+        isBootstrapMode = SystemKeyspace.getBootstrapState() == SystemKeyspace.BootstrapState.IN_PROGRESS;
```

Comment:
this is one of the critical lines: some assertions based on isBootstrapMode, like read verb handler: https://github.com/palantir/cassandra/blob/2bcaf4adf07309a1cc9c11da03f44136204d6cf6/src/java/org/apache/cassandra/db/ReadVerbHandler.java#L37 may not reject reads otherwise, as `isBootstrapMode` would be null instead of `true` on second startup

Reply:
Nit: there is an existing function for this check: `SystemKeyspace.bootstrapInProgress()`